src/main/java/kg/jarkyn/server/Incoming/RequestParser.java

```diff
@@ -0,0 +1,36 @@
+package kg.jarkyn.server.Incoming;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class RequestParser {
+
+    public static ParsedRequest parse(InputStream inputStream) {
```

Comment:
I do quite like the functional approach of this class. You have functions with an input and output, and there's no need to instantiate a RequestParser as it's just a bag of functions.

If you do feel like you have too many private static methods, you can internally instantiate a private RequestParser and call methods on that.. for e.g.

` return new RequestParser(inputStream).parse()`  

Then you'd only have the single public static method.